client/src/components/header.jsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import Button from '@mui/material/Button';
+import Dropdown from './dropdown';
+import '../page/homepage.css'
+const Header = function () {
+
+    const teacherLinks = [{name: "Teacher Absences", url: "/teacherAbsences"},
+                         {name: "Teacher", url: "/cvghjiop"}]
+                         
+return (  
+<div className='header'>
+
+    <Dropdown dropdownlist={teacherLinks} root="Teachers"> </Dropdown>
```

Comment:
These dropdowns do not close when another is opened, meaning a dropdown can be visible but not clickable. Can you add a property `open` or `selected` to these dropdowns, then use a state variable to hold the last clicked dropdown? That way a dropdown can close automatically if another is clicked

e.g. (my rough guess at how this would work):
```
//In header.jsx
const [selectedDropdown, setSelectedDropdown] = useState(null)
...
<Dropdown onChange={x => setSelectedDropdown(x)} open={selectedDropdown === "Teachers"} dropdownlist={teacherLinks} root="Teachers" />
//In dropdown.jsx
// Call onChange(root) on click
// Call onChange(null) on close
```

Reply:
technical debt, should probably be a low priority bug fix ticket